packages/ui-utils/src/createChainedFunction.ts

```diff
@@ -38,8 +38,8 @@ type GenericFunction = (...args: any[]) => any
  * @returns {function|null}
  */
 
-function createChainedFunction(
-  ...funcs: (null | undefined | GenericFunction)[]
+function createChainedFunction<F extends GenericFunction = GenericFunction>(
```

Comment:
I added this to ensure that all functions passed have the same/compatible types.

Reply:
oh this is cool :)